src/OpenTelemetry/Logs/LogRecord.cs

```diff
@@ -58,20 +59,39 @@ internal LogRecord()
             Exception? exception,
             IReadOnlyList<KeyValuePair<string, object?>>? stateValues)
         {
-            this.Data = new(Activity.Current)
+            var activity = Activity.Current;
+
+            this.Data = new(activity)
             {
                 TimestampBacking = timestamp,
+                Severity = (LogRecordSeverity)logLevel,
+            };
 
+            this.ILoggerData = new()
+            {
+                TraceState = activity?.TraceStateString,
                 CategoryName = categoryName,
-                LogLevel = logLevel,
+                FormattedMessage = formattedMessage,
                 EventId = eventId,
-                Message = formattedMessage,
                 Exception = exception,
+                State = state,
             };
 
+            if (stateValues != null && stateValues.Count > 0)
+            {
+                var lastAttribute = stateValues[stateValues.Count - 1];
+                this.Data.Body = lastAttribute.Key == "{OriginalFormat}"
+                   ? lastAttribute.Value as string
+                   : null;
+            }
+
+            this.Data.Body ??= formattedMessage;
+
+            this.InstrumentationScope = null;
+
+            this.Attributes = stateValues;
```

Comment:
nit: rename local `stateValues` -> `attributes`

Reply:
This ctor our code no longer uses. I'm maintaining it because I know of at least one user calling it to modify log records which were previously immutable. So I left it "stateValues" to be ultra-conservative to not break anyone's reflection. Unlikely they are looking at parameter names but I figured might as well be safe?

Reply:
> I'm maintaining it because I know of at least one user calling it to modify log records which were previously immutable.

Hmm, now that they're mutable, maybe we're ok to break them now? If so, probably makes sense to do it in a separate PR for visibility.